lib/schema_tools/modules/hash.rb

```diff
@@ -192,6 +189,16 @@ def parse_object(obj, field, prop, opts)
         res
       end
 
+      private
+
+      def parse_object?(obj, field)
+        if obj.respond_to?( field )
+          rel_obj = obj.send( field )
```

Comment:
I just wanted to a `true : false` interface, rather than an `object : nil` interface. Otherwise, your way is cleaner